From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: zcrypt: Do not remove coprocessor for error 8/72
References: bnc#575181,LTC#59763
Patch-mainline: 19b123ebacacdce5e75045bfe82122b01c821a5b

Symptom:     Coprocessor cards going offline
Solution:    Look for 8/72 error code and return -EINVAL.

Acked-by: John Jolly <jjolly@suse.de>

--- linux-2.5.orig/drivers/s390/crypto/zcrypt_pcixcc.c
+++ linux-2.5/drivers/s390/crypto/zcrypt_pcixcc.c
@@ -470,6 +470,8 @@ static int convert_type86_ica(struct zcr
 		}
 		if (service_rc == 12 && service_rs == 769)
 			return -EINVAL;
+		if (service_rc == 8 && service_rs == 72)
+			return -EINVAL;
 		zdev->online = 0;
 		return -EAGAIN;	/* repeat the request on a different device. */
 	}
--- linux-2.5.orig/drivers/s390/crypto/zcrypt_pcicc.c
+++ linux-2.5/drivers/s390/crypto/zcrypt_pcicc.c
@@ -373,6 +373,8 @@ static int convert_type86(struct zcrypt_
 			zdev->max_mod_size = PCICC_MAX_MOD_SIZE_OLD;
 			return -EAGAIN;
 		}
+		if (service_rc == 8 && service_rs == 72)
+			return -EINVAL;
 		zdev->online = 0;
 		return -EAGAIN;	/* repeat the request on a different device. */
 	}
